manifests/init.pp

```diff
@@ -497,7 +497,7 @@
   Variant[Undef,Integer,Pattern[/^(\d+)$/]] $rabbitmq_port = undef,
   Optional[String]   $rabbitmq_host = undef,
   Optional[String]   $rabbitmq_user = undef,
-  Optional[String]   $rabbitmq_password = undef,
+  Variant[Undef,Sensitive,String] $rabbitmq_password = undef,
```

Comment:
I think that you are seeing the sensitive info in the diff's is because the type is incorrect. Try `Variant[Undef, Sensitive]`

https://docs.puppet.com/puppet/4.10/lang_data_sensitive.html